controlplane/kubeadm/internal/controllers/helpers_test.go

```diff
@@ -237,11 +237,22 @@ func TestReconcileKubeconfigSecretDoesNotAdoptsUserSecrets(t *testing.T) {
 		},
 	}
```

Comment:
Q: I don't have a good overview. Are we covering the relevant cases with the unit tests?

Reply:
Added test cases covering the new ownerRef changing behaviour now

Reply:
Thx!